### PR TITLE
fix __@observable

### PR DESCRIPTION
**Pull request by evmar**
This fixes the rxjs sync.  See individual commits.

### CODE REVIEW

File: src/type_translator.ts [hunk 1]
```diff
@@ -9,6 +9,25 @@
...
+ *     'complex name': string;
+ *   }
+ *   Foo.bar;  // ok
+ *   Foo['bar']  // ok
+ *   Foo['complex name']  // ok
+ *
+ * In Closure-land, we want identify that the legal name 'bar' can become an
+ * ordinary field, but we need to skip strings like 'complex name'.
+ */
+export function isValidClosurePropertyName(name: string): boolean {
```
Comment (evmar) [inline src/type_translator.ts]: This is a weird place for this function.  But its previous home was in a jsdoc file which is also a weird location.  I don't feel great about it.

File: test_files/unique_symbol/unique_symbol.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// library.
+declare global {
+  interface SymbolConstructor {
+    readonly observable: symbol;
+  }
+}
+
+// This test is verifying the type of this expression, which ultimately
+// refers to some TypeScript internal __@observeable thing.
+export let x: {[Symbol.observable]: string}|null = null;
```
Comment (mprobst) [inline test_files/unique_symbol/unique_symbol.ts]: can you give this one more property? That'd make it more obvious to me that it's only filtering the bad ones.

By the way, I think this also fixes another problem:

```
const x: {'quoted-thing': string}: {...};
```

Maybe add a test for that so we don't regress to just specially treating symbols? In that case, I think it'd be nice to call this test folder something like non identifier properties or so.

[the remark this answers is not in the document]
Comment (evmar) [reply test_files/unique_symbol/unique_symbol.ts]: Done

Comment (evmar) [reply test_files/unique_symbol/unique_symbol.ts]: Done

**Pull request merged**